File: BetterChannelIcons/src/main/java/xyz/wingio/plugins/betterchannelicons/PluginSettings.java
package xyz.wingio.plugins.betterchannelicons;

import android.content.Context;
import android.view.*;
import android.widget.*;

import androidx.recyclerview.widget.*;
import androidx.core.content.ContextCompat;

import xyz.wingio.plugins.BetterChannelIcons;
import xyz.wingio.plugins.betterchannelicons.recycler.*;
import xyz.wingio.plugins.betterchannelicons.*;

import com.aliucord.Utils;
import com.aliucord.utils.*;
import com.aliucord.api.SettingsAPI;
import com.aliucord.fragments.SettingsPage;
import com.aliucord.views.Button;
import com.aliucord.views.ToolbarButton;

import com.lytefast.flexinput.R;

import java.util.*;

public class PluginSettings extends SettingsPage {
    private SettingsAPI settings;

    public PluginSettings(SettingsAPI settings){
        this.settings = settings;
    }

    private final int settingsId = View.generateViewId();

    @Override
    @SuppressWarnings("ResultOfMethodCallIgnored")
    public void onViewBound(View view) {
        super.onViewBound(view);
        Context ctx = view.getContext();
        LinearLayout layout = getLinearLayout();
        setActionBarTitle("BetterChannelIcons");

        RecyclerView recyclerView = new RecyclerView(ctx);
        recyclerView.setLayoutManager(new LinearLayoutManager(ctx));
        Map<String, String> icons = settings.getObject("icons", new HashMap<>(), xyz.wingio.plugins.betterchannelicons.Utils.iconStoreType);
        
        
        recyclerView.setAdapter(new IconListAdapter(this, icons));

        Button newIcon = new Button(ctx);
        newIcon.setText("Add Icon");
        newIcon.setOnClickListener(v -> {
            new AddChannelSheet(this, settings).show(getFragmentManager(), "add_channel_sheet");
        });

        layout.addView(newIcon);
        layout.addView(recyclerView);

        LinearLayout toolbarButtons = new LinearLayout(ctx);
        toolbarButtons.setOrientation(LinearLayout.HORIZONTAL);
        toolbarButtons.setLayoutParams(new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT));

        LinearLayout.LayoutParams marginEndParams = new LinearLayout.LayoutParams(Toolbar.LayoutParams.WRAP_CONTENT, Toolbar.LayoutParams.WRAP_CONTENT);
        toolbarButtons.setHorizontalGravity(Gravity.END);
        marginEndParams.setMarginEnd(DimenUtils.getDefaultPadding());
        ToolbarButton settingsBtn = new ToolbarButton(ctx);
        settingsBtn.setLayoutParams(marginEndParams);
        settingsBtn.setImageDrawable(ContextCompat.getDrawable(ctx, R.d.ic_guild_settings_24dp));
        toolbarButtons.setId(settingsId);
        toolbarButtons.addView(settingsBtn);

        settingsBtn.setOnClickListener(e -> {
            new SettingsSheet(this, settings).show(getParentFragmentManager(), "Settings");
        });
        ViewGroup toolbar = (ViewGroup) getHeaderBar();
        if(toolbar.findViewById(settingsId) == null) toolbar.addView(toolbarButtons);
    }
}

File: BetterChannelIcons/src/main/java/xyz/wingio/plugins/betterchannelicons/PluginSettings.kt
@file:Suppress("MISSING_DEPENDENCY_CLASS", "MISSING_DEPENDENCY_SUPERCLASS")

package xyz.wingio.plugins.betterchannelicons

import android.annotation.SuppressLint
import android.view.Gravity
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.Toolbar

import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

import com.aliucord.api.SettingsAPI
import com.aliucord.fragments.SettingsPage
import com.aliucord.utils.DimenUtils.defaultPadding
import com.aliucord.views.Button
import com.aliucord.views.ToolbarButton

import com.lytefast.flexinput.R

import xyz.wingio.plugins.betterchannelicons.recycler.IconListAdapter

class PluginSettings(private val settings: SettingsAPI) : SettingsPage() {

    private val settingsId = View.generateViewId()

    @SuppressLint("SetTextI18n")
    override fun onViewBound(view: View) {
        super.onViewBound(view)
        val ctx = view.context
        val layout = linearLayout
        val icons: MutableMap<String, String> = settings.getObject("icons", HashMap(), Utils.iconStoreType)

        setActionBarTitle("BetterChannelIcons")

        Button(ctx).apply {
            text = "Add Icon"

            setOnClickListener {
                AddChannelSheet(this@PluginSettings, settings).show(
                    parentFragmentManager, "add_channel_sheet"
                )
            }
            layout.addView(this)
        }

        RecyclerView(ctx).apply {
            layoutManager = LinearLayoutManager(ctx)
            adapter = IconListAdapter(this@PluginSettings, icons)

            layout.addView(this)
        }

        val toolbarButtons = LinearLayout(ctx).apply {
            id = settingsId
            orientation = LinearLayout.HORIZONTAL
            layoutParams = LinearLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.WRAP_CONTENT
            )

            setHorizontalGravity(Gravity.END)

            addView(
                ToolbarButton(ctx).apply {
                    layoutParams = LinearLayout.LayoutParams(
                        Toolbar.LayoutParams.WRAP_CONTENT,
                        Toolbar.LayoutParams.WRAP_CONTENT
                    ).apply { marginEnd = defaultPadding }

                    setImageDrawable(ContextCompat.getDrawable(ctx, R.e.ic_guild_settings_24dp))
                    setOnClickListener {
                        SettingsSheet(settings).show(parentFragmentManager, "Settings")
                    }
                }
            )
        }

        val toolbar = headerBar as ViewGroup
        if (toolbar.findViewById<View?>(settingsId) == null) toolbar.addView(toolbarButtons)
    }

}